src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/ServiceHubLifetimeManagerFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal class ServiceHubLifetimeManagerFactory
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ServiceManagerContext _context;
+        public ServiceHubLifetimeManagerFactory(IServiceProvider sp, IOptions<ServiceManagerContext> context)
+        {
+            _serviceProvider = sp;
+            _context = context.Value;
+        }
+        public async Task<IServiceHubLifetimeManager> Create(string hubName, CancellationToken cancellationToken, ILoggerFactory loggerFactoryPerHub=null)
+        {
+            switch (_context.ServiceTransportType)
+            {
+                case ServiceTransportType.Persistent:
+                    {
+                        var containerFactory = _serviceProvider.GetRequiredService<MultiEndpointConnectionContainerFactory>();
```

Comment:
no need to call `GetRequiredService` manually? put `MultiEndpointConnectionContainerFactory` into ctor?